src/core.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    reproject(input_data, output_projection; shape_out = nothing, order = 1, hdu_in = 1, hdu_out = 1)
+
+Reprojects image data to a new projection using interpolation.
+
+# Arguments
+- `input_data`: Image data which is being reprojected.
+                It can be an ImageHDU, FITS object or name of a FITS file.
+- `output_projection`: Frame in which data is reprojected.
+                       Frame can be taken from WCSTransform object, ImageHDU, FITS or name of FITS file.
+- `shape_out`: Shape of image after reprojection.
+- `order`: Order of interpolation.
+           0: Nearest-neighbor
+           1: Linear
+           2: Quadratic
+- `hdu_in`: Used to specify HDU number when giving input as FITS or name of FITS file.
+- `hud_out:` Used to specify HDU number when giving output projection as FITS or name of FITS file.
+"""
+function reproject(input_data, output_projection; shape_out = nothing, order::Int = 1, hdu_in::Int = 1, hdu_out::Int = 1)::Tuple{Array{Float64,2},BitArray{2}}
```

Comment:
Is the type signature of the output really necessary for type stability?  I almost never need to use it.